test/runner/lib/target.py

```diff
@@ -376,6 +377,23 @@ def analyze_integration_target_dependencies(integration_targets):
                         if hidden_target_name in meta_line:
                             dependencies[hidden_target_name].add(role_target.name)
 
+    while True:
+        changes = 0
+
+        for dummy, dependent_target_names in dependencies.items():
+            for dependent_target_name in list(dependent_target_names):
+                new_target_names = dependencies.get(dependent_target_name)
+
+                if new_target_names:
+                    for new_target_name in new_target_names:
+                        if new_target_name not in dependent_target_names:
+                            dependent_target_names.add(new_target_name)
+                            changes += 1
+                    dependent_target_names.remove(dependent_target_name)
+
+        if not changes:
+            break
```

Comment:
Note: If the only way you end up using changes is as a boolean condition it's better to just use boolean values.  ie: instead of making it an integer and incrementing, do this:
``` python
changes = False
[...]
if new_targete_name not in dependent_target_names:
    changes = True
[...]
if not changes:
    break
```